src/core_plugins/metrics/public/lib/html_id_generator.js

```diff
@@ -0,0 +1,12 @@
+import uuid from 'uuid';
+
+/**
+ * This function returns a function to generate ids.
+ * This can be used to generate unique, but predictable ids to pair labels
+ * with their inputs. It takes an optional prefix as a parameter. If you don't
+ * specify it, it generates a random id prefix.
+ */
+export function htmlIdGenerator(idPrefix) {
+  const prefix = idPrefix || uuid.v1();
+  return (suffix) => `${prefix}_${suffix}`;
```

Comment:
I'd consider changing the implementation slightly so any call to the id-generator function produces actually a unique-id. Right now, it may produce duplicate ids.

```
const generator = htmlIdGenerator();
const fooUniqueId = generator();
const barUniqueId = generator();
//fooUniqueId === barUniqueId
```

Reply:
Good point. Will change this.